tools/kochdocs.nim

```diff
@@ -283,10 +155,127 @@ lib/wrappers/openssl.nim
+lib/pure/collections/setimpl.nim
+""".splitWhitespace() # these don't produce any useful info even with `nim doc0`
+
+proc isRelativeTo(path, base: string): bool =
+  # pending #13212 use os.isRelativeTo
+  let path = path.normalizedPath
+  let base = base.normalizedPath
+  let ret = relativePath(path, base)
+  result = path.len > 0 and not ret.startsWith ".."
+
+proc getDocList(): seq[string] =
+  var t: HashSet[string]
+  for a in doc0:
+    doAssert a notin t
+    t.incl a
+  for a in withoutIndex:
+    doAssert a notin t, a
+    t.incl a
+
+  for a in ignoredModules:
+    doAssert a notin t, a
+    t.incl a
+
+  var t2: HashSet[string]
+  template myadd(a)=
+    result.add a
+    doAssert a notin t2, a
+    t2.incl a
+
+  # don't ignore these even though in lib/system
+  const goodSystem = """
+lib/system/io.nim
+lib/system/nimscript.nim
+lib/system/assertions.nim
+lib/system/iterators.nim
+lib/system/dollars.nim
+lib/system/widestrs.nim
+""".splitWhitespace()
+
+  for a in walkDirRec("lib"):
+    if a.splitFile.ext != ".nim": continue
+    if a.isRelativeTo("lib/pure/includes"): continue
+    if a.isRelativeTo("lib/genode"): continue
+    if a.isRelativeTo("lib/deprecated"):
+      if a notin @["lib/deprecated/pure/ospaths.nim"]: # REMOVE
+        continue
+    if a.isRelativeTo("lib/system"):
+      if a notin goodSystem:
+        continue
+    if a notin t:
+      result.add a
+      doAssert a notin t2, a
+      t2.incl a
+
+  myadd "nimsuggest/sexp.nim"
+  # these are include files, even though some of them don't specify `included from ...`
+  const ignore = """
+compiler/ccgcalls.nim
+compiler/ccgexprs.nim
+compiler/ccgliterals.nim
+compiler/ccgstmts.nim
+compiler/ccgthreadvars.nim
+compiler/ccgtrav.nim
+compiler/ccgtypes.nim
+compiler/jstypes.nim
+compiler/semcall.nim
+compiler/semexprs.nim
+compiler/semfields.nim
+compiler/semgnrc.nim
+compiler/seminst.nim
+compiler/semmagic.nim
+compiler/semobjconstr.nim
+compiler/semstmts.nim
+compiler/semtempl.nim
+compiler/semtypes.nim
+compiler/sizealignoffsetimpl.nim
+compiler/suggest.nim
+compiler/packagehandling.nim
+compiler/hlo.nim
+compiler/rodimpl.nim
+compiler/vmops.nim
+compiler/vmhooks.nim
+""".splitWhitespace()
+
+  # not include files but doesn't work; not included/imported anywhere; dead code?
+  const bad = """
+compiler/debuginfo.nim
+compiler/canonicalizer.nim
+compiler/forloops.nim
+""".splitWhitespace()
+
+  # these cause errors even though they're imported (some of which are mysterious)
+  const bad2 = """
+compiler/closureiters.nim
+compiler/tccgen.nim
+compiler/lambdalifting.nim
+compiler/layouter.nim
+compiler/evalffi.nim
+compiler/nimfix/nimfix.nim
+compiler/plugins/active.nim
+compiler/plugins/itersgen.nim
+""".splitWhitespace()
+
+  for a in walkDirRec("compiler"):
+    if a.splitFile.ext != ".nim": continue
+    if a in ignore: continue
+    if a in bad: continue
+    if a in bad2: continue
+    result.add a
+
+const doc = getDocList()
```

Comment:
But I don't want this list to be computed at compile-time! koch is a tool, I rarely recompile it, yet the doc generation process should be able to pick up new modules...

Reply:
good point, fixed